File: solutions/typescript/secret-handshake/1/secret-handshake.ts
export function commands(n: number): string[] {
  const binary = n.toString(2);
  const binaryArr = binary.split('').reverse();

  const actions = [];

  if (binaryArr[0] === '1') actions.push('wink');
  if (binaryArr[1] === '1') actions.push('double blink');
  if (binaryArr[2] === '1') actions.push('close your eyes');
  if (binaryArr[3] === '1') actions.push('jump')
  if (binaryArr[4] === '1') actions.reverse();
  
  return actions;
}

File: solutions/python/secret-handshake/1/secret_handshake.py
def commands(binary_str):
    actions = ['wink', 'double blink', 'close your eyes', 'jump']
    
    handshake = [action for action, digit in zip(actions, [*binary_str][:0:-1]) if digit == '1']
    
    return handshake[::-1] if binary_str[0] == '1' else handshake
        
